src/VisionModel/resnet.jl

```diff
@@ -0,0 +1,72 @@
+using Flux
+
+basicblock(inplanes::Int, outplanes::Int, downsample::Bool = false) = downsample ? 
+  Chain(Conv((3, 3), inplanes => outplanes, stride = 1, pad = 1),
+        BatchNorm(outplanes, λ = relu),
+        Conv((3, 3), outplanes => outplanes, stride = 2, pad = 1,
+        BatchNorm(outplanes, λ=relu)) : 
+  Chain(Conv((3, 3), inplanes => outplanes, stride = 1, pad = 1),
+        BatchNorm(outplanes, λ = relu),
+        Conv((3, 3), outplanes => outplanes, stride = 1, pad = 1,
+        BatchNorm(outplanes, λ=relu))
+
+bottleneck(inplanes::Int, outplanes::Int, downsample::Bool = false) = downsample ?
+  Chain(Conv((1, 1), inplanes => inplanes, stride = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((3, 3), inplanes => inplanes, stride = 1, pad = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((1, 1), inplanes => outplanes, stride = 2),
+        BatchNorm(outplanes, λ = relu)) :
+  Chain(Conv((1, 1), inplanes => inplanes, stride = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((3, 3), inplanes => inplanes, stride = 1, pad = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((1, 1), inplanes => outplanes, stride = 1),
+        BatchNorm(outplanes, λ = relu))
```

Comment:
I realized that I made a significant error that we left uncaught. We've been looping over `channel_config` in the main loop, but the expansion is handled within the block functions. Additionally, there is a corner case that we've been ignoring. Take ResNet-50 as an example. The first bottleneck is repeated 3 times. The first repetition takes in 64 channels, and the bottleneck block outputs [64, 64, 256] channels (respectively for each convolution). The shortcut also takes in 64 channels and outputs 256 channels. The final result is 256 channels. The second bottleneck in the repetition receives 256 channels instead of 64.

So, to deal with these two issues, I suggest we modify the block functions like so:
```julia
bottleneck(inplanes::Int, outchannels, downsample::Bool = false) = downsample ?
   Chain(Conv((1, 1), inplanes => outchannels[1], stride = 2),
         BatchNorm(inplanes, λ = relu),
         Conv((3, 3), inplanes => outchannels[2], stride = 1, pad = 1),
         BatchNorm(inplanes, λ = relu),
         Conv((1, 1), inplanes => outchannels[3], stride = 1),
         BatchNorm(outplanes, λ = relu)) :
   Chain(Conv((1, 1), inplanes => outchannels[1], stride = 1),
         BatchNorm(inplanes, λ = relu),
         Conv((3, 3), inplanes => outchannels[2], stride = 1, pad = 1),
         BatchNorm(inplanes, λ = relu),
         Conv((1, 1), inplanes => outchannels[3], stride = 1),
         BatchNorm(outplanes, λ = relu))
```

I also made the slight modification of moving the `stride = 2` to the first `Conv`, since that's where downsampling is done in the paper. We would make the same changes for `basicblock` for consistency.

Reply:
Normally, the 1st ``Conv`` layer of ``bottleneck`` is ``inplanes => inplanes`` but since ``outchannels[1]`` has same value it really doesn't make any difference. I understand it now.

Reply:
any changes to ``identity`` or ``projection`` functions?

Reply:
There were some slight errors in my code above, so I edited the comment.

Yes, we will need to modify `identity` to perform zero-padding whenever `inchannels != outchannels`.